website/archiver/model.py

```diff
@@ -0,0 +1,112 @@
+from copy import deepcopy
+import datetime
+
+from modularodm import (
+    fields,
+)
+
+from framework.mongo import ObjectId
+from framework.mongo import StoredObject
+
+from website.archiver import (
+    ARCHIVER_INITIATED,
+    ARCHIVER_SUCCESS,
+    ARCHIVER_FAILURE,
+    ARCHIVER_FAILURE_STATUSES
+)
+
+from website.addons.base import StorageAddonBase
+from website import settings
+
+class ArchiveJob(StoredObject):
+
+    _id = fields.StringField(
+        primary=True,
+        default=lambda: str(ObjectId())
+    )
+
+    done = fields.BooleanField(default=False)
+    sent = fields.BooleanField(default=False)
+    status = fields.StringField()
+    datetime_initiated = fields.DateTimeField(default=datetime.datetime.utcnow)
+
+    dst_node = fields.ForeignField('node', backref='active')
+    src_node = fields.ForeignField('node')
+    initiator = fields.ForeignField('user')
+
+    # Dictonary mapping addon short name to archive status
+    # {
+    #   [addon_short_name]: {
+    #     'status': [STATUS_CONSTANT],
+    #     'errrors': []
+    #   }
+    # }
+    target_addons = fields.DictionaryField(default={})
+
+    @property
+    def children(self):
+        return [node.archive_job for node in self.dst_node.nodes]
+
+    @property
+    def parent(self):
+        parent_node = self.dst_node.parent_node
+        return parent_node.archive_job if parent_node else None
+
+    @property
+    def success(self):
+        return self.status == ARCHIVER_SUCCESS
+
+    def info(self):
```

Comment:
This method seems unnecessary. Why not just use attribute access when you need these?

If you really need a serialization of these specific attributes, it might make more sense to return a dictionary rather than a tuple.